File: tests/test-terraria/src/main/java/pize/tests/terraria/map/MapTile.java
package pize.tests.terraria.map;

import pize.tests.terraria.tile.TileType;

public class MapTile{

    private TileType type;
    private byte light;

    public MapTile(TileType type, int light){
        setType(type);
        setLight(light);
    }


    public byte getLight(){
        return light;
    }

    public void setLight(int light){
        this.light = (byte) light;
    }


    public TileType getType(){
        return type;
    }

    public void setType(TileType type){
        this.type = type;
    }

}

File: tests/test-terraria/src/main/kotlin/pize/tests/terraria/map/MapTile.kt
package pize.tests.terraria.map

import pize.tests.terraria.tile.TileType

class MapTile(type: TileType?, light: Int) {
    var type: TileType? = null
    var light: Byte = 0
        private set

    init {
        this.type = type
        setLight(light)
    }

    fun setLight(light: Int) {
        this.light = light.toByte()
    }
}
